client/state/happychat/selectors.js

```diff
@@ -89,3 +92,12 @@ export const canUserSendMessages = createSelector(
 	),
 	[ getHappychatConnectionStatus, getHappychatChatStatus ]
 );
+
+export const getHappychatLastActivityTimestamp = state => state.happychat.lastActivityTimestamp;
+
+export const wasHappychatRecentlyActive = state => {
```

Comment:
Ah yes, I should have added a comment to that commit.

It was really a defensive measure - there's no real reason the `lastActive` timestamp should be in the future. If it is, it means we have invalid data and should ignore it (eg, maybe lastActive was accidentally persisted in milliseconds instead of seconds, in which case `wasHappychatRecentlyActive` would evaluate to true for a looong time).

In a previous deploy which attempted to implement the middleware refactor in one big go, this recently active check was missing and far too many Calypso clients connected, DDoSing the service. So you can probably understand my extra defensiveness here :)